var/spack/repos/builtin/packages/lzo/package.py

```diff
@@ -37,15 +37,14 @@ class Lzo(Package):
     version('2.06', '95380bd4081f85ef08c5209f4107e9f8')
     version('2.05', 'c67cda5fa191bab761c7cb06fe091e36')
 
-    def install(self, spec, prefix):
-        configure_args = [
-            '--prefix={0}'.format(prefix),
+    def configure_args(self):
+        return [
             '--disable-dependency-tracking',
             '--enable-shared'
         ]
-        configure(*configure_args)
-        make()
-        if self.run_tests:
-            make('check')
-            make('test')  # more exhaustive test
-        make('install')
+
+    @AutotoolsPackage.sanity_check('build')
+    @AutotoolsPackage.on_package_attributes(run_tests=True)
+    def check(self):
```

Comment:
I agree with @adamjstewart -- it would be good if package authors didn't have to add decorators that control the build pipeline flow.  They should just have to know the right name to override...  is that doable?